hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/metrics/ProvidesMetrics.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core.metrics;
+
+/**
+ * Marker interface to be implemented by all classes that want to be able
```

Comment:
Comment from Viliam: `ProvidesMetrics` is not technically a "marker interface". Marker interface has no methods. `Serializable` is a marker interface, for example.

Reply:
True, but I think hiding the method would just make the code more awkward and not calling it a "marker interface" would just make it harder to understand/explain. I would not change anything. @cangencer pls. comment.